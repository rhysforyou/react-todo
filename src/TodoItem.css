.todo-list-item {
  border-top: 1px solid #ddd;
  box-sizing: border-box;
  font-family: sans-serif;
  color: #333;
  list-style: none;
  padding: 1rem;
}

.todo-list-item:hover {
  background: ghostwhite;
}

.todo-list-item input {
  margin-right: 1rem;
}

.todo-list-item.completed {
  color: #999;
  text-decoration: line-through;
}
