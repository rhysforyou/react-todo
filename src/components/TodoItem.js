import './TodoItem.css'
import PropTypes from 'prop-types';
import React from 'react'

const completionClass = (completed) => completed ? 'completed' : 'incomplete'
const TodoItem = ({title, completed, onClick}) => (
  <li className={`todo-list-item ${completionClass(completed)}`}
    onClick={onClick}>
    <label>
      <input type='checkbox'
        defaultChecked={completed} />
      {title}
    </label>
  </li>
)
TodoItem.propTypes = {
  title: PropTypes.string.isRequired,
  completed: PropTypes.bool.isRequired,
  onClick: PropTypes.func
}

export default TodoItem