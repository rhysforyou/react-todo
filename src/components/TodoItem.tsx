import './TodoItem.css'
import React from 'react'

const completionClass = (completed: boolean) =>
  completed ? 'completed' : 'incomplete'

interface TodoItemProps {
  title: string
  completed: boolean
  onClick: () => void
}

const TodoItem = ({ title, completed, onClick }: TodoItemProps) => (
  <li className={`todo-list-item ${completionClass(completed)}`}>
    <label>
      <input
        type="checkbox"
        checked={completed}
        onChange={e => {
          onClick()
        }}
      />
      {title}
    </label>
  </li>
)

export default TodoItem
