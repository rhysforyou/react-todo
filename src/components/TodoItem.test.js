/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoItem from './TodoItem'

Enzyme.configure({ adapter: new Adapter() });

describe('TodoItem', () => {
  it('forwards click events to its `onClick` property', () => {
    const onClick = jest.fn()
    const wrapper = Enzyme.shallow(
      <TodoItem title='Test Item'
        completed={false}
        onClick={onClick} />
    )
    wrapper.find('.todo-list-item').simulate('click')
    expect(onClick).toBeCalled()
  })

  describe('`completion` property', () => {
    it("applies the 'completed' class if present", () => {
      const wrapper = Enzyme.shallow(
        <TodoItem title='Test Item'
          completed={true} />
      )
      expect(wrapper.find('.completed'))
    })

    it("applies the 'incomplete' class if absent", () => {
      const wrapper = Enzyme.shallow(
        <TodoItem title='Test Item'
          completed={false} />
      )
      expect(wrapper.find('.incomplet'))
    })
  })
})
