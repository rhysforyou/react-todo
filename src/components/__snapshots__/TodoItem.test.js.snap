exports[`test renders correctly 1`] = `
<li
  className="todo-list-item completed"
  onClick={undefined}>
  <label>
    <input
      checked={true}
      type="checkbox" />
    Test Item
  </label>
</li>
`;
